From f2e6a7be477924fcd3978633b4f0cbb34091b5fe Mon Sep 17 00:00:00 2001
From: Hector Cao <hector.cao@canonical.com>
Date: Sat, 26 Aug 2023 14:16:45 +0200
Subject: [PATCH] Fix qemu compilation for Xen

Xen uses qemu i386, tdx patchset add kvm_get_vm_type() in target/i386/kvm/kvm.c
whereas Xen uses the stub kvm file (target/i386/kvm/kvm-stub.c) that does not
contains this function definition. This causes a link error for Xen.
To fix that error, we should add this function definition in the kvm stub
file also.
---
 target/i386/kvm/kvm-stub.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/target/i386/kvm/kvm-stub.c b/target/i386/kvm/kvm-stub.c
index e052f1c7b0..1e923e61a9 100644
--- a/target/i386/kvm/kvm-stub.c
+++ b/target/i386/kvm/kvm-stub.c
@@ -13,6 +13,9 @@
 #include "cpu.h"
 #include "kvm_i386.h"
 
+// For KVM_X86_DEFAULT_VM definition
+#include <linux/kvm.h>
+
 #ifndef __OPTIMIZE__
 bool kvm_has_smm(void)
 {
@@ -49,3 +52,8 @@ void kvm_set_max_apic_id(uint32_t max_apic_id)
 {
     return;
 }
+
+int kvm_get_vm_type(MachineState *ms, const char *vm_type)
+{
+    return KVM_X86_DEFAULT_VM;
+}
-- 
2.39.2

